### PR TITLE
fix: proxy-nsmd-k8s always filters for local NSEs

### PR DESCRIPTION
**Pull request by denis-tingajkin**
Signed-off-by: Denis Tingajkin <denis.tingajkin@xored.com>

## Motivation and Context
https://github.com/networkservicemesh/networkservicemesh/issues/2202

## How Has This Been Tested?

- [X] Covered by existing integration testing
- [ ] Added integration testing to cover
- [ ] Tested locally
- [ ] Have not tested

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by edwarnicke** [comment]
@tiswanso Does this look like what you have in mind?

File: k8s/pkg/proxyregistryserver/discovery.go [hunk 1]
```diff
@@ -70,9 +70,14 @@ func (d *discoveryService) FindNetworkService(ctx context.Context, request *regi
...
 			return nil, dErr
 		}
-		for _, nsm := range response.NetworkServiceManagers {
+		managers := make(map[string]*registry.NetworkServiceManager)
+		for key, nsm := range response.NetworkServiceManagers {
+			managers[key] = nsm
 			if url, urlErr := d.currentDomainNSMgrURL(ctx, d.clusterInfoService, nsm.Url); urlErr == nil && nsm.Url == url {
-				return d.handleLocalFindCase(response, url), nil
+				d.handleLocalFindCase(response, nsm, url)
+				delete(managers, key)
```
Comment (tiswanso) [inline k8s/pkg/proxyregistryserver/discovery.go:78]: I think you could just move the add (line 75) to after the if statement and remove this delete, right?

**Comment by tiswanso** [comment]
> @tiswanso Does this look like what you have in mind?

@edwarnicke  -- if I understand it correctly, it looks good.  Looks like it just localizes the network service manager entries (for local NSEs), in which case, it preserves the full list of NSEs.  The local nsmgr info is an improvement.

**Comment by denis-tingajkin** [comment]
@tiswanso yes, you are correct. This PR just add possible to localize local nsmgrs/nses. This behavior is more correct then just replace all nsmgrs/nses to local if it is actually local case https://github.com/networkservicemesh/networkservicemesh/issues/2169 and it is fixing https://github.com/networkservicemesh/networkservicemesh/issues/2202

File: k8s/pkg/proxyregistryserver/discovery.go (hunk 1 above)
Comment (denis-tingajkin) [reply k8s/pkg/proxyregistryserver/discovery.go]: Thanks, fixed.

**Comment by tiswanso** [comment]
fyi, this fix works for me in my setup.  Thanks, @denis-tingajkin !

**Pull request merged**